src/utils/arkTransaction.ts

```diff
@@ -166,3 +168,159 @@ const nLocktimeMinSeconds = 500_000_000n;
 function isSeconds(locktime: bigint): boolean {
     return locktime >= nLocktimeMinSeconds;
 }
+
+/**
+ * Verify tapscript signatures on a transaction input
+ * @param tx Transaction to verify
+ * @param inputIndex Index of the input to verify
+ * @param requiredSigners List of required signer pubkeys (hex encoded)
+ * @param excludePubkeys List of pubkeys to exclude from verification (hex encoded, e.g., server key not yet signed)
+ * @param allowedSighashTypes List of allowed sighash types (defaults to [SigHash.DEFAULT])
+ * @throws Error if verification fails
+ */
+export function verifyTapscriptSignatures(
+    tx: Transaction,
+    inputIndex: number,
+    requiredSigners: string[],
+    excludePubkeys: string[] = [],
+    allowedSighashTypes: number[] = [SigHash.DEFAULT]
+): void {
```

Comment:
```suggestion
export function verifyTapscriptSignatures(
    prevoutGetter: (o: Outpoint) => TransactionOutput
    tx: Transaction,
    inputIndex: number,
    requiredSigners: string[],
    excludePubkeys: string[] = [],
    allowedSighashTypes: number[] = [SigHash.DEFAULT]
): void {
```

Should we add a `prevoutGetter` in order to not rely on witnessUtxo fields while computing the message to sign ? mainly for 2 reasons:
* allow verifying some flow that are not setting witnessUtxo field in PSBT
* allow to pass a "prevoutGetter" fetching prevouts from external sources (explorer, db...). thus you do not have to trust the party who set the witnessUtxo fields

Reply:
Yeah I agree, both LGTM but having an explicit fetcher may avoid ppl to run the validation without proper verification of the witnessUtxo ?

Reply:
EH, I'd leave that to other sanity check elsewhere, would rather keep this lean